spec/facility_spec.rb

```diff
@@ -1,26 +1,230 @@
+      @facility_1.add_service('Vehicle Registration')
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      @bolt = Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev} )
+      @camaro = Vehicle.new({vin: '1a2b3c4d5e6f', year: 1969, make: 'Chevrolet', model: 'Camaro', engine: :ice} )
+    end
+    context 'adding vehicles to the registered vehicle array' do
+      it 'will add registered Vehicles to the array' do
+        @facility_1.register_vehicle(@cruz)
+        expect(@facility_1.registered_vehicles).to eq([@cruz])
+        @facility_1.register_vehicle(@bolt)
+        expect(@facility_1.registered_vehicles).to eq([@cruz, @bolt])
+      end
+      it 'will collect a fee when registering vehicles' do
+        @facility_1.register_vehicle(@cruz)
+        expect(@facility_1.collected_fees).to eq(100)
+      end
+      it 'will collect a fee of $100 if the plate type is :regular' do
+        @facility_1.register_vehicle(@cruz)
+        expect(@facility_1.collected_fees).to eq(100)
+      end
+      it 'will collect a feee of $200 if the plate type is :ev' do
+        @facility_1.register_vehicle(@bolt)
+        expect(@facility_1.collected_fees).to eq(200)
+      end
+      it 'will collect a feee of $25 if the plate type is :antique' do
+        @facility_1.register_vehicle(@camaro)
+        expect(@facility_1.collected_fees).to eq(25)
+      end
+    end
+
+    context 'changes made to Vehicles when registering' do
+      before(:each) do
+        @facility_1.add_service('Vehicle Registration')
+        @facility_1.register_vehicle(@cruz)
+        @facility_1.register_vehicle(@bolt)
+        @facility_1.register_vehicle(@camaro)
+      end
+      it 'assigns the current date as a vehicles registration date' do
+        expect(@cruz.registration_date).to be_an_instance_of(Date)
+        expect(@cruz.registration_date).to eq(Date.today)
+      end
+      it 'assigns a plate type based' do
+        expect(@cruz.plate_type).to eq(:regular)
+      end
+      it 'assigns an antique plate if a vehicle is more than 25 years old' do
+        expect(@camaro.plate_type).to eq(:antique)
+      end
+      it 'assings an ev plate if a vehicle has an ev engine type' do
+        expect(@bolt.plate_type).to eq(:ev)
+      end
+    end
+  end 
+  
+  describe '#administer written test' do
+    before(:each) do
+      @registrant_1 = Registrant.new('Bruce', 18, true )
+      @registrant_2 = Registrant.new('Penny', 16 )
+      @registrant_3 = Registrant.new('Tucker', 15 )
+    end
+    context 'when adminstering written tests to qualified registrants' do
+      before(:each) { @facility_1.add_service('Written Test')}
+      it 'Will change a registrants license data to show written as true' do
+        expect(@facility_1.administer_written_test(@registrant_1)).to eq(true)
+        expect(@registrant_1.license_data[:written]).to eq(true)
```

Comment:
For tests like this one, consider adding an assertion before the execution method call (in this case `administer_written_test`. That way, the test clearly demonstrates that the state starts out with one value and is changed after the relevant call.
```
expect(@registrant_1.license_data[:written]).to eq(false)
expect(@facility_1.administer_written_test(@registrant_1)).to eq(true)
expect(@registrant_1.license_data[:written]).to eq(true)
```